manila/network/neutron/neutron_network_plugin.py

```diff
@@ -301,6 +305,20 @@ def deallocate_network(self, context, share_server_id):
         for port in ports:
             self._delete_port(context, port)
 
+        # It may be possible that there are ports existing without a
+        # corresponding manila network allocation entry in the manila db,
+        # because port create request may have been successfully sent to
+        # neutron, but the response, the created port could not be stored
+        # in manila due to unreachable db
+        if share_network_subnet:
+            ports = self.neutron_api.list_ports(
+                network_id=share_network_subnet['neutron_net_id'],
+                device_owner='manila:share',
+                device_id=share_server_id)
+
+            for port in ports:
+                self._delete_port(context, port)
```

Comment:
we have in mind that the port may not be in manila db, soft delete it might fail?

Reply:
the intention is to delete the ports that are in manila db as they are getting deleted today. But after that if there is possible leak of ports,(i.e. we dont have entries in db), we ask Neutron. And then delete those ports if any returned by Neutron.
Please note, if ports from db are deleted, neutron will only return the ports that are indeed belongs to share server and not present in db.

Reply:
> neutron will only return the ports that are indeed belongs to share server and not present in db

exactly, and if there are any such ports left (to be deleted in the second loop here) for those the `self.db.network_allocation_delete(context, port['id'])` from the `else` part in `_delete_port()` will fail :(

Reply:
fixed